### PR TITLE
make the custom route controller use server-side-apply to avoid stomping unknown fields

### PR DESCRIPTION
**Pull request by deads2k**
Only the specified fields will be written.  The Force will overwrite any other actor trying to set these fields.

/hold

need to hold to ensure the ComponentRouteStatus slice apply annotations are correct.

/assign @stlaz @ibihim 

@Miciah this is the delivery of a promised debt fix from 4.11.
@tkashem @JoelSpeed this model should be used when operators share joint ownership of fields, versus control the authoritative copy.

### CODE REVIEW

**Comment by deads2k** [comment]
I think we'll need https://github.com/openshift/api/pull/1262 before merging this one.

**Comment by deads2k** [comment]
TestCustomRouterCerts passed in https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-authentication-operator/582/pull-ci-openshift-cluster-authentication-operator-master-e2e-operator/1562146522651430912

**Comment by JoelSpeed** [comment]
Code changes look good, happy to label once the API PR is fixed up and merged

File: pkg/controllers/customroute/custom_route_controller.go
Comment (tkashem) [inline pkg/controllers/customroute/custom_route_controller.go:264]: @deads2k  is it safe to assume that SSA is always enabled on the server?

Comment (deads2k) [reply pkg/controllers/customroute/custom_route_controller.go:264]: > @deads2k is it safe to assume that SSA is always enabled on the server?

Yes

**Comment by deads2k** [comment]
looks to have worked

/hold cancel

File: pkg/controllers/customroute/custom_route_controller.go
Comment (stlaz) [inline pkg/controllers/customroute/custom_route_controller.go:9]: keep the previous order of imports please

**Comment by deads2k** [comment]
comments addressed.

**Review by ibihim — APPROVED** [review]
/lgtm

**Pull request merged**
